ggml/src/ggml-cpu/ggml-cpu-aarch64.c

```diff
@@ -1090,466 +1090,62 @@ void ggml_gemm_q4_0_4x4_q8_0(int n, float * restrict s, size_t bs, const void *
     UNUSED(ncols_interleaved);
     UNUSED(blocklen);
 
-#if ! ((defined(_MSC_VER)) && ! defined(__clang__)) && defined(__aarch64__) && defined(__ARM_NEON)
+#if ! ((defined(_MSC_VER)) && ! defined(__clang__)) && defined(__aarch64__) && defined(__ARM_NEON) && defined(__ARM_FEATURE_DOTPROD)
     if (ggml_cpu_has_neon() && ggml_cpu_has_dotprod()) {
-        const void * b_ptr = vx;
-        const void * a_ptr = vy;
-        float * res_ptr = s;
-        size_t res_stride = bs * sizeof(float);
+        const block_q8_0x4 * a_ptr = (const block_q8_0x4 *) vy;
+        for (int r = 0; r < nr; r += 4) {
+            const block_iq4_nlx4 * b_ptr = (const block_iq4_nlx4 *) vx;
```

Comment:
```suggestion
            const block_q4_0x4 * b_ptr = (const block_q4_0x4 *) vx;
```

Reply:
Oh good catch! Funny that all my tests were good with that 😱